### PR TITLE
Define initial project structure with basic java entities

### PR DESCRIPTION
**Pull request by cristian0x**
Basic java entities definition with simple GET and POST endpoints to verify if it works correctly.

For task: "Define database structure and java entitites"

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -33,7 +33,7 @@
 		<dependency>
 			<groupId>org.postgresql</groupId>
 			<artifactId>postgresql</artifactId>
-			<scope>runtime</scope>
+<!--			<scope>runtime</scope>-->
```
Comment (bartekbaran) [inline pom.xml]: To remove

File: src/main/java/com/gigacal/config/PersistenceConfig.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.gigacal.config;
+
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+
+@EnableJpaRepositories(basePackages = "com.gigacal.repository")
+public class PersistenceConfig {
```
Comment (bartekbaran) [inline src/main/java/com/gigacal/config/PersistenceConfig.java]: Perhaps this class can also be removed - can you check if everything is working as intended without it?

File: src/main/java/com/gigacal/entity/AuditEntity.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import jakarta.persistence.Column;
+import jakarta.persistence.MappedSuperclass;
+import lombok.Data;
+import java.time.LocalDateTime;
+
+@MappedSuperclass
+@Data
+public class AuditEntity {
+
+    @Column(updatable = false)
```
Comment (bartekbaran) [inline src/main/java/com/gigacal/entity/AuditEntity.java]: Can we also add nullable = false to the Entity fields that are required?

**Review by bartekbaran — APPROVED** [review]
Fix the comments and it's good to go

File: src/main/java/com/gigacal/config/PersistenceConfig.java (hunk 2 above)
Comment (cristian0x) [reply src/main/java/com/gigacal/config/PersistenceConfig.java]: Removed, everything works fine without it.

**Pull request merged**